modules/face/samples/sample_face_swapping.cpp

```diff
@@ -0,0 +1,195 @@
+    vector<int> ind(3);
+    for( size_t i = 0; i < triangleList.size(); i++ )
+    {
+        Vec6f triangle = triangleList[i];
+        pt[0] = Point2f(triangle[0], triangle[1]);
+        pt[1] = Point2f(triangle[2], triangle[3]);
+        pt[2] = Point2f(triangle[4], triangle[5]);
+        if ( rect.contains(pt[0]) && rect.contains(pt[1]) && rect.contains(pt[2])){
+            for(int j = 0; j < 3; j++)
+                for(size_t k = 0; k < points.size(); k++)
+                    if(abs(pt[j].x - points[k].x) < 1.0 && abs(pt[j].y - points[k].y) < 1)
+                        ind[j] =(int) k;
+            delaunayTri.push_back(ind);
+        }
+    }
+}
+// Warps and alpha blends triangular regions from img1 and img2 to img
+void warpTriangle(Mat &img1, Mat &img2, vector<Point2f> &t1, vector<Point2f> &t2)
+{
+    Rect r1 = boundingRect(t1);
+    Rect r2 = boundingRect(t2);
+    // Offset points by left top corner of the respective rectangles
+    vector<Point2f> t1Rect, t2Rect;
+    vector<Point> t2RectInt;
+    for(int i = 0; i < 3; i++)
+    {
+        t1Rect.push_back( Point2f( t1[i].x - r1.x, t1[i].y -  r1.y) );
+        t2Rect.push_back( Point2f( t2[i].x - r2.x, t2[i].y - r2.y) );
+        t2RectInt.push_back( Point((int)(t2[i].x - r2.x),(int) (t2[i].y - r2.y))); // for fillConvexPoly
+    }
+    // Get mask by filling triangle
+    Mat mask = Mat::zeros(r2.height, r2.width, CV_32FC3);
+    fillConvexPoly(mask, t2RectInt, Scalar(1.0, 1.0, 1.0), 16, 0);
+    // Apply warpImage to small rectangular patches
+    Mat img1Rect;
+    img1(r1).copyTo(img1Rect);
+    Mat img2Rect = Mat::zeros(r2.height, r2.width, img1Rect.type());
+    applyAffineTransform(img2Rect, img1Rect, t1Rect, t2Rect);
+    multiply(img2Rect,mask, img2Rect);
+    multiply(img2(r2), Scalar(1.0,1.0,1.0) - mask, img2(r2));
+    img2(r2) = img2(r2) + img2Rect;
+}
+int main( int argc, char** argv)
+{
+    //Give the path to the directory containing all the files containing data
+    CommandLineParser parser(argc, argv,
+        "{ help h usage ? |      | give the following arguments in following format }"
+        "{ image1 i1      |      | (required) path to the first image file in which you want to apply swapping }"
+        "{ image2 i2      |      | (required) path to the second image file in which you want to apply face swapping }"
+        "{ model m        |      | (required) path to the file containing model to be loaded for face landmark detection}"
+        "{ face_cascade f |      | Path to the face cascade xml file which you want to use as a detector}"
+    );
+    // Read in the input arguments
+    if (parser.has("help")){
+        parser.printMessage();
+        cerr << "TIP: Use absolute paths to avoid any problems with the software!" << endl;
+        return 0;
+    }
+    string image1(parser.get<string>("image1"));
+    string image2(parser.get<string>("image2"));
+    Mat img1=imread(image1);
+    Mat img2=imread(image2);
+    if (img1.empty()){
```

Comment:
Sir, I thought of informing the user, which image is not present, hence added two if's. Should I remove it ?